load_into_pg.py

```diff
@@ -394,16 +394,23 @@ def moveTableToSchema(table, schemaName, dbConnectionParam):
 # load a project
 elif args.so_project:
     import libarchive
+    import tempfile
 
     filepath = None
     if args.file:
         filepath = args.file
+        url = filepath
     else:
-        # download the 7z archive in /tmp
+        # download the 7z archive in tempdir
         file_name = args.so_project + '.stackexchange.com.7z'
         url = '{0}/{1}'.format(args.archive_url, file_name)
-        filepath = '/tmp/'+file_name
-        six.print_('Downloading the archive, please be patient ...')
+        temp_dir = tempfile.gettempdir()
+        if temp_dir == 'None':
```

Comment:
You probably meant `if temp_dir is None` instead of doing a string comparison here.

Also, I would recommend using [`temp_dir = mkdtemp()`](https://docs.python.org/2/library/tempfile.html#tempfile.mkdtemp), which would prevent accidental overwriting of the file in the `tmp` folder. This could lead to subtle bugs, e.g. if the `/tmp` is shared (e.g. a computer lab connected to an NFS) by multiple users (e.g. students at the computer lab) who attempt to download the same file simultaneously.

Reply:
you are totally right for the comparison.
No problem to use mkdtemp()

Reply:
is it safe to remove all check on the mkdtemp() result?

Reply:
Good question; I'm not sure. Let me check.

Reply:
It seems like [`mdktemp`](https://github.com/python/cpython/blob/master/Lib/tempfile.py#L333) will raise exceptions if it fails, and only return a valid directly if it returns at all.

Note that we'll have to delete the directory ourselves in case the user doesn't want to keep the file around.

Reply:
Last commit should solve questions around temp directories and final deletion.
Does it look good to you?